nlp/2021_06_18_data_augmentation/totw_nlp_dat_aug.ipynb

```diff
@@ -0,0 +1,1789 @@
+        "    return [el['translation_text'] for el in twohopback_translated_texts]\n",
+        "    \n",
+        "backtranslate_dataset = lambda dataset: dataset.map(lambda x: {'text': back_tranlation_nl_fr_en_nl(x[\"text\"])}, batch_size=10, batched=True)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "4gGKYuM0Sx05"
+      },
+      "source": [
+        "# Back-translate the training dataset\n",
+        "book_review_train_ds_back = backtranslate_dataset(book_review_train_ds)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "fQIw3q-DSE9J"
+      },
+      "source": [
+        "### ✨ Contextual word embedding insertions\n",
+        "\n",
+        "\n",
+        "The [nlpaug](https://github.com/makcedward/nlpaug) library combines frequently used augmentation techniques into a python package. We'll use the `ContextualWordEmbsForSentenceAug` component that uses contextual word embeddings to find the top n similar words for augmentation. \n",
+        "\n",
+        "The contextual embeddings are retrieved from the tranformer-based pretrained RoBERTa model, which was trained on the Dutch section of the [OSCAR](https://oscar-corpus.com/) corpus. The word embeddings have a dependence on the surrounding words, this defines the **context** of the embededing.  "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "_mj-2mXiVKIR"
+      },
+      "source": [
+        "aug = naf.Sequential([\n",
+        "    naw.ContextualWordEmbsAug(\n",
+        "        model_path='pdelobelle/robbert-v2-dutch-base',\n",
+        "        model_type='roberta',\n",
+        "        aug_p=0.20,\n",
+        "        action=\"insert\")\n",
+        "])\n",
+        "\n",
+        "replace_newline = lambda dataset: dataset.map(lambda x: {'text': x[\"text\"].replace(\"\\n\",' ')}, batched=False)\n",
+        "contextual_emb_aug = lambda dataset: dataset.map(lambda x: {'text': aug.augment(x[\"text\"])},  batch_size=10, batched=True)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "background_save": true
+        },
+        "id": "2gfA_jeOStEN"
+      },
+      "source": [
+        "# Removing newlines in the text and performing word insertions based on contextual word embeddings\n",
+        "book_review_train_ds_newline = replace_newline(book_review_train_ds)\n",
+        "book_review_train_ds_contemb = contextual_emb_aug(book_review_train_ds_newline)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "hjKUMX0VSS_T"
+      },
+      "source": [
+        "### 💪 Combination of both techniques\n",
```

Comment:
Maybe we could add one more to this list, but just as an honorable mention: augmentation using a transformer-based paraphrasing model, such as: https://github.com/PrithivirajDamodaran/Parrot_Paraphraser

Reply:
But with the sidenote that we weren't able to test it out since our problem focuses on dutch